File: DAM_1/DAM_2º-año_1º/Programación/UD2-Ej1/src/Ej4.java
import java.util.Scanner;

/**
 * Realiza un programa que pida al usuario introducir un número y devuelva su raíz
 * cuadrada. Ayuda: la clase Math tiene una función (sqrt) que calcula la raíz cuadrada.
 * **/
public class Ej4 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Dime un número para darte su raíz: ");
        int numero = scanner.nextInt();

        double raiz = Math.sqrt(numero);

        System.out.println("La raíz de " + numero + " es: " + raiz);
    }
}

File: DAM_1/DAM_2º-año_1º/Programación/UD7 - Kotlin Ej1/src/Ej4.kt
/**
 * Dado un número, muestra si es primo.
 */

class Ej4 {
    fun esPrimo(n: Int): Boolean {
        if (n <= 1) return false
        if (n == 2) return true
        if (n % 2 == 0) return false
        for (i in 3..Math.sqrt(n.toDouble()).toInt() step 2) {
            if (n % i == 0) return false
        }
        return true
    }

}
fun main(){
    val ej4 = Ej4();

    print("Dime un número para saber si es primo o no: ")
    val numero = readln().toInt();

    if (ej4.esPrimo(numero)){
        print("El número es primo");
    }else{
        print("El número no es primo");
    }

}